ui/Main.elm

```diff
@@ -0,0 +1,341 @@
+module Main exposing (..)
+
+import Set
+import Kinto
+import Json.Decode as Decode
+import Html
+import Html.Attributes
+import Html.Events
+
+
+type alias SystemAddon =
+    { id : String
+    , version : String
+    }
+
+
+type alias SystemAddonVersions =
+    { id : String
+    , builtin : Maybe String
+    , update : Maybe String
+    }
+
+
+type alias ReleaseDetails =
+    { buildId : String
+    , channel : String
+    , filename : String
+    , lang : String
+    , target : String
+    , url : String
+    , version : String
+    }
+
+
+type alias Channel =
+    String
+
+
+type alias Release =
+    { builtins : List SystemAddon
+    , updates : Maybe (List SystemAddon)
```

Comment:
Because the go script converts empty lists to null when serializing json from what I understood from @leplatrem.

Refraining from commenting further about Go.